internal/test/ouroboros_mock/connection.go

```diff
@@ -187,16 +187,21 @@ func (c *Connection) processInputEntry(entry ConversationEntry) error {
 		if msg == nil {
 			return fmt.Errorf("received unknown message type: %d", msgType)
 		}
-		// Set CBOR for expected message to match received to make comparison easier
-		entry.InputMessage.SetCbor(msg.Cbor())
-		// Compare received message to expected message
+
+		// Compare received message to expected message, excluding the cbor content
+		//
+		// As changing the CBOR of the expected message is not thread-safe, we instead temporarily
+		// clear the CBOR of the received message
+		msgCbor := msg.Cbor()
+		msg.ClearCbor()
 		if !reflect.DeepEqual(msg, entry.InputMessage) {
 			return fmt.Errorf(
 				"parsed message does not match expected value: got %#v, expected %#v",
 				msg,
 				entry.InputMessage,
 			)
 		}
+		msg.SetCbor(msgCbor)
```

Comment:
Is this needed? We don't use `msg` again in this function

Reply:
removed